### PR TITLE
Stricter entity property checking for floats and integers

### PR DESCRIPTION
**Pull request by Xterionix**
Needs #267 to be fixed before it can be merged.

### CODE REVIEW

File: src/Lib/Diagnostics/BehaviorPack/Entity/properties.ts [hunk 1]
```diff
@@ -207,12 +207,22 @@ function check_entity_property_usage(
...
+        `${parent}/${name}/${value}`,
+        `Property value is not a integer: ${value}`,
+        DiagnosticSeverity.error,
+        `behaviorpack.entity.property.${definition.type}.value`
+      );
+      break;
+
     case "float":
-      if (typeof value === "number" || typeof value === "string") return;
+      if ((typeof value === "number" && !Number.isInteger(value)) || typeof value === "string") return;
```
Comment (DaanV2) [inline src/Lib/Diagnostics/BehaviorPack/Entity/properties.ts:221]: Im not sure for this extra checks on floats 🤔 

cause 3 and 3.2 are still both valid floats

Comment (Xterionix) [reply src/Lib/Diagnostics/BehaviorPack/Entity/properties.ts:221]: I mean you can test it in game if you'd like; they've carried this over to items too in 1.21.40.
> Numbers and booleans are no longer interchangeable in the JSON input
> Floating point numbers are no longer accepted where an integer is expected

Comment (DaanV2) [reply src/Lib/Diagnostics/BehaviorPack/Entity/properties.ts:221]: Thats says floats are not accepted as integer. My comment is about integers being used in floats

Comment (Xterionix) [reply src/Lib/Diagnostics/BehaviorPack/Entity/properties.ts:221]: Vice versa is true for actor properties. You can test it yourself if you'd like
image

Comment (DaanV2) [reply src/Lib/Diagnostics/BehaviorPack/Entity/properties.ts:221]: What I mean is the following:

```json
"paeaefae:aeae": {
    "type": "float",
    "default": 3.2,
    "range": [ 0.0, 2.0 ]
}

"paeaefae:aeae": {
    "type": "float",
    "default": 3,
    "range": [ 0.0, 2.0 ]
}
```

both of these need to be valid, with your introduction of `!Number.IsInteger` the second case will fail

Comment (Xterionix) [reply src/Lib/Diagnostics/BehaviorPack/Entity/properties.ts:221]: @DaanV2

Comment (DaanV2) [reply src/Lib/Diagnostics/BehaviorPack/Entity/properties.ts:221]: Going to need to validate this myself in game, but I'm quite busy this week

**Review by DaanV2 — APPROVED** [review]

**Pull request merged**